### PR TITLE
[TFLM] Added optimized softmax kernel (int32 and float) for CEVA-BX1 and CEVA SP500

### PR DESCRIPTION
**Pull request by yair-ehrenwald**
Relevant github issue:https://github.com/tensorflow/tensorflow/issues/45607

### CODE REVIEW

File: tensorflow/lite/micro/kernels/ceva/softmax.cc [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/softmax.h"
+#ifdef MCPS_MEASUREMENT
+#include "tensorflow/lite/micro/kernels/ceva/mcps_macros.h"
+#endif
+
+#if defined(CEVA_BX1) || defined(CEVA_SP500)
+extern int32_t* CEVA_TFLM_KERNELS_SCRATCH;
+extern int32_t CEVA_TFLM_KERNELS_SCRATCH_SIZE_VAL;
+#endif
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/ceva/softmax.cc:35]: worth putting these in a ceva_common.h or something similar to avoid repeating the same include and extern in each of the kernels?

File: tensorflow/lite/micro/kernels/ceva/softmax.cc [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      "Test params:Call CEVA_TFLM_Softmax_Float32 %d times, inetrnal loop = %d",
+      outer_size_mcps, depth_mcps);
+#endif
+}
+
+TfLiteStatus SoftmaxQuantizedCEVA(TfLiteContext* context,
+                                  const TfLiteEvalTensor* input,
+                                  TfLiteEvalTensor* output,
+                                  const SoftmaxParams& op_data) {
+  if (input->type == kTfLiteUInt8) {
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/ceva/softmax.cc:75]: drop uint8 support?

**Review by advaitjain — CHANGES_REQUESTED** [review]
small comments only.

Happy to make a PR that removes uint8 support from the reference kernel by this Friday.

File: tensorflow/lite/micro/kernels/ceva/softmax.cc (hunk 2 above)
Comment (yair-ehrenwald) [reply tensorflow/lite/micro/kernels/ceva/softmax.cc]: Done

File: tensorflow/lite/micro/kernels/ceva/softmax.cc [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+#include "tensorflow/lite/micro/kernels/ceva/ceva_common.h"
+#include "tensorflow/lite/micro/kernels/ceva/ceva_tflm_lib.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/softmax.h"
+#ifdef MCPS_MEASUREMENT
+#include "tensorflow/lite/micro/kernels/ceva/mcps_macros.h"
+#endif
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/ceva/softmax.cc:31]: you could add these to ceva_common.h as well?

No change needed in this PR. If you think it is worthwhile then you can make the change in all the kernels in a follow-up PR.

**Review by advaitjain — APPROVED** [review]

File: tensorflow/lite/micro/kernels/ceva/softmax.cc (hunk 3 above)
Comment (yair-ehrenwald) [reply tensorflow/lite/micro/kernels/ceva/softmax.cc:31]: I'd like to have our measurement macros in all kernels, but only some of them require scratch memory, so I'll leave it as is for the time being. Eventually I think we will do some work on the whole scratch usage issue, exact size calculations per kernel etc and hopefully come up with something maintainable.
